diff --git a/kernel/manager/apk_sign.c b/kernel/manager/apk_sign.c
index bc84408d..16494551 100644
--- a/kernel/manager/apk_sign.c
+++ b/kernel/manager/apk_sign.c
@@ -32,6 +32,7 @@ struct sdesc {
 
 static apk_sign_key_t apk_sign_keys[] = {
     { EXPECTED_SIZE_RESUKISU, EXPECTED_HASH_RESUKISU }, /* ReSukiSU/ReSukiSU */
+    { EXPECTED_SIZE_KITTISU, EXPECTED_HASH_KITTISU }, /* KittiSU/KittiSU */
 #ifdef CONFIG_KSU_MULTI_MANAGER_SUPPORT
     { EXPECTED_SIZE_OFFICIAL, EXPECTED_HASH_OFFICIAL }, // tiann/KernelSU
     { EXPECTED_SIZE_5EC1CFF, EXPECTED_HASH_5EC1CFF }, // 5ec1cff/KernelSU
diff --git a/kernel/manager/manager_sign.h b/kernel/manager/manager_sign.h
index 87e78338..96a03d85 100644
--- a/kernel/manager/manager_sign.h
+++ b/kernel/manager/manager_sign.h
@@ -25,6 +25,10 @@
 #define EXPECTED_SIZE_KOWX712 0x375
 #define EXPECTED_HASH_KOWX712 "484fcba6e6c43b1fb09700633bf2fb4758f13cb0b2f4457b80d075084b26c588"
 
+// KittiSU/KittiSU
+#define EXPECTED_SIZE_KITTISU 0x2e8
+#define EXPECTED_HASH_KITTISU "9a8b292b21f688282a054729049afdb99ab67f24eb10fb05102963b79ac99810"
+
 typedef struct {
     unsigned size;
     const char *sha256;
